nemo/collections/nlp/data/information_retrieval/information_retrieval_dataset.py

```diff
@@ -0,0 +1,288 @@
+        dataset_dict = {id_: tokens[:max_seq_length] for (id_, tokens) in dataset_dict}
+        return dataset_dict
+
+    def preprocess_line(self, line):
+        id_, text = line.split("\t")
+        token_ids = self.tokenizer.text_to_ids(text.strip())
+        return int(id_), token_ids
+
+    def construct_input(self, token_ids1, max_seq_length, token_ids2=None):
+        input_ids = [self.tokenizer.pad_id] * max_seq_length
+        bert_input = [self.tokenizer.cls_id] + token_ids1 + [self.tokenizer.sep_id]
+        sentence1_length = len(bert_input)
+        if token_ids2 is not None:
+            bert_input = bert_input + token_ids2 + [self.tokenizer.sep_id]
+
+        bert_input = bert_input[:max_seq_length]
+
+        num_nonpad_tokens = len(bert_input)
+
+        input_ids[:num_nonpad_tokens] = bert_input
+        input_ids = np.array(input_ids, dtype=np.long)
+        input_mask = input_ids != self.tokenizer.pad_id
+        input_type_ids = np.ones_like(input_ids)
+        input_type_ids[:sentence1_length] = 0
+
+        return input_ids, input_mask, input_type_ids
+
+    def preprocess_bert(self, query_id, psg_ids):
+        """
+        Transforms query id (Q) and a list of passages ids (P1, ..., Pk)
+        into a tensor of size [k, max_length] with the following rows:
+        [CLS] Q_text [SEP] Pi_text [SEP], i = 1, ..., k
+        """
+
+        max_seq_length = self.max_query_length + self.max_passage_length + 3
+        input_ids, input_mask, input_type_ids = [], [], []
+        for psg_id in psg_ids:
+            inputs = self.construct_input(self.queries[query_id], max_seq_length, self._psgid2tokens(psg_id))
+            input_ids.append(inputs[0])
+            input_mask.append(inputs[1])
+            input_type_ids.append(inputs[2])
+
+        input_ids = np.stack(input_ids)
+        input_mask = np.stack(input_mask)
+        input_type_ids = np.stack(input_type_ids)
+
+        return input_ids, input_mask, input_type_ids
+
+    def preprocess_dpr(self, query_id, psg_ids):
+        """
+        Transforms query id (Q) and a list of passages ids (P1, ..., Pk)
+        into two tensors of sizes [1, max_q_length] and [k, max_p_length]
+        with the following rows:
+        1) [CLS] Q_text [SEP]
+        2) [CLS] Pi_text [SEP], i = 1, ..., k
+        """
+
+        q_input_ids, q_input_mask, q_type_ids = self.construct_input(self.queries[query_id], self.max_query_length + 2)
+        input_ids, input_mask, input_type_ids = [], [], []
+        for psg_id in psg_ids:
+            inputs = self.construct_input(self._psgid2tokens(psg_id), self.max_passage_length + 2)
+            input_ids.append(inputs[0])
+            input_mask.append(inputs[1])
+            input_type_ids.append(inputs[2])
+        input_ids = np.stack(input_ids)
+        input_mask = np.stack(input_mask)
+        input_type_ids = np.stack(input_type_ids)
+        return (
+            q_input_ids[None, ...],
```

Comment:
what does this do?

Reply:
tensor[None, ...] adds new dimension (e.g., batch dimension), similar to torch.unsqueeze(0)